src/Http/Client/Response.php

```diff
@@ -442,7 +444,31 @@ public function getCookieData($name)
             return null;
         }
 
-        return $this->cookies->get($name)->toArrayClient();
+        $cookie = $this->cookies->get($name);
+
+        return $this->convertCookie($cookie);
+    }
+
+    /**
+     * Convert the cookie into an array of its properties.
+     *
+     * This method is compatible with older client code that
+     * expects date strings instead of timestamps.
+     *
+     * @param \Cake\Http\Cookie\CookieInterface $cookie Cookie object.
+     * @return array
+     */
+    public function convertCookie(CookieInterface $cookie)
```

Comment:
I made them public so I could test them easily ;). But right, that doesn't need to be piblic.